services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_compose_utils.py

```diff
@@ -37,36 +46,32 @@ def _increase_timeout(docker_command_timeout: int | None) -> int | None:
 
 
 @run_sequentially_in_context()
-async def _write_file_and_spawn_process(
-    yaml_content: str,
-    *,
-    command: str,
-    process_termination_timeout: int | None,
+async def _compose_cli_command(
+    yaml_content: str, *, command: str, process_termination_timeout: int | None
 ) -> CommandResult:
-    """The command which accepts {file_path} as an argument for string formatting
+    """
+    This calls is intentionally verbose at DEBUG level
+    """
 
-    ALL docker_compose run sequentially
+    env_vars = _DOCKER_COMPOSE_CLI_ENV
 
-    This calls is intentionally verbose at INFO level
-    """
-    async with write_to_tmp_file(yaml_content) as file_path:
-        cmd = command.format(file_path=file_path)
+    _logger.debug("Runs '%s' with ENV=%s...\n%s", command, env_vars, yaml_content)
 
-        logger.debug("Runs %s ...\n%s", cmd, yaml_content)
+    result = await async_command(
+        command=command,
+        timeout=process_termination_timeout,
+        pipe_as_input=yaml_content,
+        env_vars=env_vars,
+    )
 
-        result = await async_command(
-            command=cmd,
-            timeout=process_termination_timeout,
-        )
-        debug_message = deepcopy(result._asdict())
-        logger.debug(
-            "Finished executing docker compose command '%s' finished_ok='%s' elapsed='%s'\n%s",
-            debug_message["command"],
-            debug_message["success"],
-            debug_message["elapsed"],
-            debug_message["message"],
-        )
-        return result
+    _logger.debug(
+        "Finished executing docker compose command '%s' finished_ok='%s' elapsed='%s'\n%s",
```

Comment:
TIP:  Why not to have a function in `result`'s class that produces this type of diagnostic logs ... `CommandResult` is a very common type that we might want to log

```python
class CommandResult(NamedTuple):
    success: bool
    message: str
    command: str
    elapsed: float | None
    
    def as_log_message(self) -> str:
       return "'%s' finished_ok='%s' elapsed='%s'\n%s".format(
       self.command,
        self.success,
        self.elapsed,
        self.message,
       )
```

then wheever you use this you can
```python
_logger.debug(" Executed this things and got %s", this_thing_result.as_log_message() )
```